cmd/internal/dgather.go

```diff
@@ -361,18 +363,50 @@ func RenderJobBPM(currentJob manifest.Job, jobInstances []manifest.JobInstance,
 			}
 
 			// Parse a rendered bpm.yml into the bpm Config struct
-			jobInstances[i].BPM, err = bpm.NewConfig(bpmBytes)
+			jobIndexBPM[i], err = bpm.NewConfig(bpmBytes)
 			if err != nil {
 				return err
 			}
+			instanceBytes := generateJobInstanceSHA(currentJob, jobInstances[i])
+
+			// All instances SHA are the same, while all instances are based on index 0
+			jobInstances[i].Fingerprint = generateSHA(instanceBytes)
+		}
 
-			// Consider adding a Fingerprint to each job instance
-			// instance.Fingerprint = generateSHA(fingerPrintBytes)
+		for _, jobBPMInstance := range jobIndexBPM {
+			if !reflect.DeepEqual(jobBPMInstance, jobIndexBPM[0]) {
+				return errors.New("found different BPM job indexes, this is not supported")
+			}
 		}
+
+		currentJob.Properties.BOSHContainerization.BPM = jobIndexBPM[0]
 	}
 	return nil
 }
 
+func generateJobInstanceSHA(currentJob *manifest.Job, jobInstance manifest.JobInstance) []byte {
+	// The following is based on https://github.com/cloudfoundry-incubator/cf-operator/compare/master...jandubois:161745895-erb-rendering-using-jobs#diff-55ce7ec0a607742ace583aad21b7acbaR231
+	// Calculate a fingerprint of all config settings for this job instance
```

Comment:
Please add the "why" - would be good as a comment to the function.

Reply:
This will be addressed on the next commit.